### PR TITLE
GODRIVER-1455 switch zstd implementation to pure go

### PR DESCRIPTION
**Pull request by john-m-liu**

### CODE REVIEW

File: x/mongo/driver/compression_test.go [hunk 1]
```diff
@@ -27,25 +24,19 @@ func TestCompression(t *testing.T) {
...
 				ZlibLevel:        wiremessage.DefaultZlibLevel,
 				ZstdLevel:        wiremessage.DefaultZstdLevel,
 				UncompressedSize: int32(len(payload)),
 			}
-			compressed, err := CompressPlayoad([]byte(payload), opts)
-			if err != nil {
-				require.Contains(t, err.Error(), "support requires cgo")
-				return
-			}
+			compressed, err := CompressPlayoad(payload, opts)
```
Comment (iwysiu) [inline x/mongo/driver/compression_test.go]: CompressPlayoad -> CompressPayload.  This is in x so its ok to change.

Comment (john-m-liu) [reply x/mongo/driver/compression_test.go]: Oops, fixed

File: mongo/options/clientoptions.go [hunk 2]
```diff
@@ -356,7 +356,7 @@ func (c *ClientOptions) SetAuth(auth Credential) *ClientOptions {
 //
 // 2. "zlib" - requires server version >= 3.6
 //
-// 3. "zstd" - requires driver version >= 1.2.0, server version >= 4.2, and cgo support to be enabled.
+// 3. "zstd" - requires server version >= 4.2, and driver version >= 1.2.0 (cgo), or driver version >= 1.3.0 (without cgo)
```
Comment (john-m-liu) [inline mongo/options/clientoptions.go]: This is assuming the change is not backported, but let me know if this isn't accurate

Comment (divjotarora) [reply mongo/options/clientoptions.go]: Let's be a little more explicit. Something like `and driver version >= 1.2.0 with cgo support enabled or driver version >= 1.3.0 without cgo`.

Comment (john-m-liu) [reply mongo/options/clientoptions.go]: changed

**Comment by jxub** [comment]
That will simplify the build of the package a lot 👍

**Pull request merged**